syntax = "proto3";

package auth;


service Auth {

	rpc SignUp (SignUpRequest) returns (AuthResponse);

	rpc SignIn (SignInRequest) returns (AuthResponse);

    rpc ValidateToken (ValidateTokenRequest) returns (ValidateTokenResponse);

    rpc GetUser (GetUserRequest) returns (GetUserResponse);

    rpc GetUsers (Empty) returns (GetUsersResponse);

}

message SignUpRequest {
    string first_name = 1;
    string second_name = 2;
    string email = 3;
    string password = 4;
    string adress = 5;
    bool is_admin = 6;
}

message SignInRequest {
    string email = 1;
    string password = 2;
}

message ValidateTokenRequest {
    string token = 1;
}

message GetUserRequest {
    string uid = 1;
}

message Empty {}

message AuthResponse {
    string token = 1;
    string uid = 2;
    bool is_admin = 3;
}

message ValidateTokenResponse {
    bool valid = 1;
    string uid = 2;
}

message GetUserResponse {
    string uid = 1;
    string first_name = 2;
    string second_name = 3;
    string email = 4;
    string adress = 5;
    bool is_admin = 6;
}

message GetUsersResponse {
    repeated GetUserResponse users = 1;
}

message SuccessResponse {
    bool success = 1;
}